### PR TITLE
Hide posts from recently ignored/unfollowed contact

### PR DESCRIPTION
**Pull request by MrPetovan**
See https://friendica.mrpetovan.com/display/735a2029315ac842f7aafc0717133891

The idea is to immediately hides posts from a contact we just ignored/unfollowed. AFAIK we already do it with blocked.

Since they're only hidden, these posts will show up again if the contact is subsequently unignored/followed again.

### CODE REVIEW

**Comment by annando** [comment]
It is better to do it with the thread table. There we do have the "contact-id" table as well. And we shouldn't query constants with their numeric representations, but only by name.

**Comment by MrPetovan** [comment]
Ok for the constant, I'm not sure I understand the `contact-id` table part.

Are you saying I should only add the condition when the `thead` table is queried, or re you saying I should add a join on the `thread` table on the two queries on the `item` table?

**Comment by annando** [comment]
I'm currently unsure. Possible you should just work with the constants and we leave the rest like you did.

**Comment by MrPetovan** [comment]
Done.

**Comment by annando** [comment]
This PR could have a side effect, as I just noticed: When not disabled, we fetch full OStatus threads, even if they hadn't been started by our contacts, but by some follower.

**Comment by MrPetovan** [comment]
Is it intended? What is the use case?

**Comment by annando** [comment]
This is intended. The use case is to fetch all posts where our followers are interacting.

**Comment by MrPetovan** [comment]
What’s the solution then, limit to `thread`? Check `network` as well?

**Comment by MrPetovan** [comment]
> I know I'm a nitpicker, but I don't like some ```<``` with constants :-) I think it would be best to have something like ```rel in (?, ?)```.

Done.

**Comment by MrPetovan** [comment]
I have no idea how to do that, I'll leave it at that for now.

**Comment by annando** [comment]
I can do that, it shouldn't be that hard.

**Comment by annando** [comment]
Okay, I checked the code and it is already done this way: https://github.com/friendica/friendica/blob/develop/src/Protocol/OStatus.php#L74-L97

Could you implement the code similar? Means: ```AND (`item`.`parent-uri` != `item`.`uri` OR `contact`.`rel` IN (%s, %d))``` with ```CONTACT_IS_SHARING``` and ```CONTACT_IS_FRIEND``` as parameters?

Additionally I think we shouldn't use "readonly" here. This always appears to me like some relique from the past.

**Comment by MrPetovan** [comment]
`readonly` is set when you ignore a contact, so it is very much in use.

**Comment by annando** [comment]
Oh, didn't knew.

**Comment by MrPetovan** [comment]
> Could you implement the code similar? 

Done.

File: mod/network.php [hunk 1]
```diff
@@ -831,10 +833,12 @@ function networkThreadedView(App $a, $update, $parent)
 				(SELECT SUBSTR(`term`, 2) FROM `search` WHERE `uid` = ? AND `term` LIKE '#%') AND `otype` = ? AND `type` = ? AND `uid` = 0) AS `term`
 			ON `item`.`id` = `term`.`oid`
 			STRAIGHT_JOIN `contact` ON `contact`.`id` = `item`.`author-id`
-				AND (`item`.`parent-uri` != `item`.`uri` OR NOT (`contact`.`rel` IN (0, %d) OR `contact`.`readonly`))
+				AND (`item`.`parent-uri` != `item`.`uri` OR `contact`.`rel` IN (?, ?) AND NOT `contact`.`readonly`)
 			WHERE `item`.`uid` = 0 AND `item`.$ordering < ? AND `item`.$ordering > ?
 				AND NOT `contact`.`hidden` AND NOT `contact`.`blocked`" . $sql_tag_nets,
-			local_user(), TERM_OBJ_POST, TERM_HASHTAG, intval(CONTACT_IS_FOLLOWER), $top_limit, $bottom_limit);
+			local_user(), TERM_OBJ_POST, TERM_HASHTAG,
+			intval(CONTACT_IS_SHARING),	intval(CONTACT_IS_FRIEND),
```
Comment (annando) [inline mod/network.php]: With the new dba functions the "intval" isn't needed anymore.

**Pull request merged**